lighty-modules/lighty-aaa-aggregator/lighty-aaa/src/main/java/io/lighty/aaa/AAALighty.java

```diff
@@ -10,60 +10,197 @@
+        } else {
+            this.defaultPasswordHashService = null;
+            this.iidmStore = null;
+            this.tokenStore = null;
+            return;
+        }
     }
 
-    @Override
-    protected boolean initProcedure() throws InterruptedException {
-        final CompletableFuture<AAALightyShiroProvider> newInstance = AAALightyShiroProvider.newInstance(
-                this.dataBroker, this.aaaConfiguration, this.credentialAuth, this.server);
-        final CountDownLatch cdl = new CountDownLatch(1);
-        newInstance.whenComplete((t, u) -> {
-            AAALighty.this.aaaShiroProviderHandler.setAaaLightyShiroProvider(t);
-            cdl.countDown();
-        });
 
-        cdl.await();
-        return true;
+    private static TokenAuthenticators buildTokenAuthenticators(
+            final CredentialAuth<PasswordCredentials> credentialAuth) {
+        return new TokenAuthenticators(new HttpBasicAuth(credentialAuth));
     }
 
-    @Override
-    protected boolean stopProcedure() {
-        this.aaaShiroProviderHandler.getAaaLightyShiroProvider().close();
-        return true;
+    private void registerServletContexts(final LocalHttpServer httpService, final String moonEndpointPath) {
+
+        Preconditions.checkNotNull(httpService, "httpService cannot be null");
+        httpService.registerServlet(moonEndpointPath, new MoonTokenEndpoint(), null);
     }
 
-    private static class AAAShiroProviderHandler {
+    private void initAAAonServer() {
+        final ContextHandlerCollection contexts = new ContextHandlerCollection();
+        final ServletContextHandler mainHandler = new ServletContextHandler(contexts, "/auth", true, false);
+        final IdmLightApplication idmLightApplication = new IdmLightApplication(iidmStore, claimCache);
+        final ServletHolder idmLightServlet = new ServletHolder(new ServletContainer(ResourceConfig.forApplication(
+                idmLightApplication)));
+        idmLightServlet.setInitParameter("jersey.config.server.provider.packages",
+                "org.opendaylight.aaa.impl.provider");
+        mainHandler.addServlet(idmLightServlet, "/*");
+        server.addContextHandler(contexts);
+        this.handlers.add(contexts);
+        this.handlers.add(mainHandler);
+        this.shiroWebEnvironmentLoaderListener = new ShiroWebEnvironmentLoaderListener(shiroConfiguration,
+                dataBroker,
+                certificateManager,
+                authenticationService,
+                tokenAuthenticators,
+                tokenStore,
+                passwordHashService);
+        server.addCommonEventListener(shiroWebEnvironmentLoaderListener);
 
-        AAALightyShiroProvider aaaLightyShiroProvider;
+        final Map<String, String> properties = new HashMap<>();
+        final CustomFilterAdapterConfigurationImpl customFilterAdapterConfig =
+                new CustomFilterAdapterConfigurationImpl();
+        customFilterAdapterConfig.update(properties);
```

Comment:
Remove `properties` variable and use directly:
```java
customFilterAdapterConfig.update(new HashMap<>());
```

Reply:
fixed